executor/encoder_pl.py

```diff
@@ -0,0 +1,127 @@
+from typing import Optional
+
+import pytorch_lightning as pl
+import torch
+from torch.nn import functional as F
+from torchmetrics.functional import accuracy
+
+from .models import MeshDataModel
+
+AVAILABLE_MODELS = {
+    'PointNet-Shapenet-d1024': {
+        'model_name': 'pointnet',
+        'hidden_dim': 1024,
+        'embed_dim': 1024,
+        'model_path': '',
+    },
+    'PointConv-Shapenet-d1024': {
+        'model_name': 'pointconv',
+        'hidden_dim': 1024,
+        'embed_dim': 1024,
+        'model_path': 'https://jina-pretrained-models.s3.us-west-1.amazonaws.com/mesh_models/pointconv-shapenet-d1024.pth',
+    },
+    'PointNet-Shapenet-d512': {
+        'model_name': 'pointnet',
+        'hidden_dim': 1024,
+        'embed_dim': 512,
+        'model_path': '',
+    },
+    'PointConv-Shapenet-d512': {
+        'model_name': 'pointconv',
+        'hidden_dim': 1024,
+        'embed_dim': 512,
+        'model_path': 'https://jina-pretrained-models.s3.us-west-1.amazonaws.com/mesh_models/pointconv-shapenet-d512.pth',
+    },
+}
+
+
+class MeshDataEncoderPL(pl.LightningModule):
```

Comment:
Move this class into `datasets/models` since this is only a training script, not a executor